### ISSUE TITLE
** ALERT: If you can see this, this repository is no longer relevant. **

### ISSUE THREAD

**Issue by i336** [dz0ny/lightdm-login-chromeos#54]
First of all, to the owner of this project: please rename the title of this issue (I think you can edit issue titles) if this issue changes in the future.

This is just a quick headsup from a random user (not affiliated with Google, or this project) to say that this method/approach used by the code in this repo will no longer work.

Let me explain how/why (or you can skip to the end for the TL;DR).

If you look in [debian/postinst](https://github.com/dz0ny/lightdm-login-chromeos/blob/master/debian/postinst), you'll see that it downloads a build of Chrome from some obscure location at Google.

I'm not running Debian, so the package infrastructure in this repo won't work for me. There's nothing stopping me from trying to run it all manually, though, right?

```
$ curl http://commondatastorage.googleapis.com/chromium-browser-snapshots/Linux_ChromiumOS/LAST_CHANGE; echo
252451
```

Hey, look at that, it gave me something back! Okay, let's try the 2nd command...

```
$ wget http://commondatastorage.googleapis.com/chromium-browser-snapshots/Linux_ChromiumOS/252451/chrome-linux.zip
...
HTTP request sent, awaiting response... 200 OK
Length: 85203211 (81M) [application/zip]
...
```

And it downloaded!

I'll spare you the rest of the process - the confusion about why the `chrome` binary inside was linked to `libudev.so.0` and not `libudev.so.1` (it's a debug/test build, it should work sanely on most Linux systems like regular chromium builds do) - and I'll fast-forward to the "...ooh." part:

```
$ ls -lh chrome-linux.zip 
-rw-r--r-- 1 i336 i336 82M Feb 21  2014 chrome-linux.zip
```

See it yet? ***The ZIP file is over two years old.***

**TL;DR: The upstream sources (from Google) that this whole project is built around are no longer being updated.**

It seems that the only way you can experience Chromium OS on Linux is to build it yourself. https://www.chromium.org/developers/how-tos/build-instructions-chromeos

I might try it sometime. It can't be that hard... right? :P

**Comment by JVApen** [comment]
Hi all,

This github project is indeed dead, so it should actually be removed.
Yes, the instructions on https://www.chromium.org/developers/how-tos/build-instructions-chromeos are indeed the best ones to use. I've even created a blogspot about it chromiumosonlinux.blogspot.com, for those that are interested in it.

@dz0ny Would be be possible to reflect the current state of this project on the README page? (Or just remove the project?)

**Comment by i336** [comment]
I guess opening this issue is the next best thing, while the author is inactive. It's not immediately noticeable, but the "Issues" tab is where people leave feedback, so hopefully they see this if they go to ask/inquire.

Great to know that I found the best link. :D

I had a quick look through your blogspot (http://chromiumosonlinux.blogspot.com) - it looks pretty interesting.

I'm personally wondering if it would be possible to get the official upstream Chromium OS debug/test build images to run stably in chroot jails (Docker!) on arbitrary Linux systems (Arch, Ubuntu, ...) and their associated configuration idiosyncrasies, whatever those are.

That way, we wouldn't need to fuss over maintaining a build server to rebuild copies of Chromium OS as per the webpage you confirmed - even if we didn't have to pay for it, for example if the OpenSuSE build service accepted the project - and the result would be a lot more immersive.

My own interest is in being able to explore/play with the Ash UI; I tend to use Chrome a lot on one specific screen, and I'm wondering whether having "Chromium OS" on that screen would be more fun and/or useful than just normal Chrome. 

Besides that, running the Chromium OS UI on top of another distro would be useful to provide a self-contained/locked-down desktop that was (relatively) consistent and predictable and could be handed-off to for example family members who just want to poke around the Web; and because it's running on top of a standard Linux distro, it would be fairly trivial to configure standard remote management like `x11vnc` and keep other tools updated as well.

I had a look inside a Chromium OS image (from http://chromium.arnoldthebat.co.uk/) and I see the `chrome` binary inside links to X11 and XCB, which gives me a lot of confidence that Chrome OS does internally use X11, meaning that getting a chroot setup working should, in theory, not be *too* difficult.

It's a huge pity that `--open-ash` doesn't produce segfaults on production builds of Chrome; the code for the windowmanager and everything is baked into every binary, and yet it's horribly crashey. I've gotten it working in the past, but it produces insta-crashes when I try to use it nowadays. I wonder if the motivation to keep it flakey is politically motivated, or simply an artifact of the way the build process works; I'm considering opening a bug to ask what the deal is with this.

**Comment by NoodleCommits** [comment]
There are some "Linux_ChromiumOS_Full" builds on https://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Linux_ChromiumOS_Full/ that is actually active today.

I've made a fork of this project and updated the "postinst" file in the "DEBIAN" directory to download these builds (In the master branch.).
